src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/WidgetPageObject.java

```diff
@@ -88,43 +101,56 @@ public boolean areAllValidSwappedForIFrames() {
     return result;
   }
 
+  /**
+   * Make sure that all tags defined in the widget tags field are loaded.
+   */
   public boolean areLoaded() {
     boolean result = true;
-    int i = 1;
-    List<WebElement> widgetIFrameList = getWidgetIFrameList();
-
-    if (widgetIFrameList.isEmpty()) {
-      result = false;
-    } else {
-      for (WebElement widgetIFrame : widgetIFrameList) {
-        if (!isWidgetVisible(widgetIFrame, getWidgetBody())) {
-          result = false;
-          PageObjectLogging.log(getTagName() + " #" + i, MercuryMessages.VISIBLE_MSG, result);
-          break;
-        }
-        PageObjectLogging.log(getTagName() + " #" + i++, MercuryMessages.VISIBLE_MSG, result);
+    String[] tags = getTags();
+    for (int i = 0; i < tags.length; i++) {
+      if (!isWidgetVisible(i)) {
```

Comment:
We will keep in mind the better PageObject pattern consistency so that next time, when interracting with multiple instances of the same functionality on webpage, we will be creating separate page objects for each of the new instances.